pkg/reconcilers/deploymentconfig.go

```diff
@@ -55,18 +54,24 @@ func GenericBackendMutators() []DCMutateFn {
 		DeploymentConfigPriorityClassMutator,
 		DeploymentConfigTopologySpreadConstraintsMutator,
 		DeploymentConfigPodTemplateAnnotationsMutator,
-		DeploymentConfigEnvMutator,
 	}
 }
 
 func DeploymentConfigEnvMutator(desired, existing *appsv1.DeploymentConfig) (bool, error) {
 	update := false
-	if desired.Name == component.BackendWorkerName {
-		if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env) {
-			diff := cmp.Diff(existing.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env)
-			log.Info(fmt.Sprintf("%s spec.template.spec.containers[0].Env has changed: %s", common.ObjectInfo(desired), diff))
-			existing.Spec.Template.Spec.Containers[0].Env = desired.Spec.Template.Spec.Containers[0].Env
-			update = true
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env) {
+		diff := cmp.Diff(existing.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env)
+		log.Info(fmt.Sprintf("%s spec.template.spec.containers[0].Env has changed: %s", common.ObjectInfo(desired), diff))
+		for _, envVar := range existing.Spec.Template.Spec.Containers[0].Env {
+			if envVar.Name == "CONFIG_REDIS_ASYNC" {
+				return update, nil
```

Comment:
If we are returning false update here and the CONFIG_REDIS_ASYNC has been set manually to 0, we are not updating it back to 1. Since we want the async mode on at all times I think this needs further refactor.

Reply:
Good point ,  but are we then overriding the users desire to disable async ? 
Default async should be enabled at all time , but you can override it by manually changing the env var.

Not sure on this one, 
- I kind of side with, sensible defaults and the ability to override them.  
- On the other hand if user has enabled hpa then disabling async may cause difficulties. 

Maybe documentation that the env var is set by default to on and can be unset but should not be if hpa is enabled

Or logic to reconcile the change out . 

Not to hard to implement either way . :thinking:

Reply:
Think I am leaning the other way now 
The operator will set this back to "1" for your own good. Docs changes might not avert a potential issue with pre-existing env vars set to "0"

Reply:
I think that given there's no real benefit of having it in sync mode, enforcing async mode is good to have

Reply:
Added the extra logic , and updated the verification steps.